code-challenges.js

```diff
@@ -7,18 +7,54 @@ var secretCodeWord1 = "lackadaisical"
 var secretCodeWord2 = "gobbledygook"
 // Expected output: "g0bbl3dyg00k"
 
+//make a function that takes in a string
 
+const coder = (string) => {
+    //split the string into an array
+    let array = string.split("")
+    //map over the array and evaluate each value to return the coded character
+    let replacement = array.map(val => {
+        if (val === "a"){
+            return "4"
+        } else if (val === "e"){
+            return "3"
+        } else if (val === "i"){
+            return "1"
+        } else if (val === "o"){
+            return "0"
+        } else {
+            return val
+        }
+    //return replacement joined back into a string
+    })
+    return replacement.join("")
+}
 
-
+console.log(coder(secretCodeWord1))
+console.log(coder(secretCodeWord2))
 
 // --------------------2) Create a function that takes in an array and returns all the words that have the letter a in them.
 
 var arrayOfWords = ["Apple", "Banana", "Plum", "Cherry", "Kiwi", "Peach"]
 // Expected output: "Apple" "Banana" "Peach"
 
+//create a function that takes in an array
+const aFruityFinder = (array) => {
+    //create a new array to push a-fruits into
+    let newArray = []
+    //map over the array
+    array.map(value => {
+        //if the string-value lowercased contains, an "a"
+        if (value.toLowerCase().includes("a")){
+            //push the value into the new array
+            newArray.push(value)
+        }
+    })
+    //return the new array of "a" containers
```

Comment:
if you use a filter instead of a map, you could omit having to push to newArray ;)